src/functions/GroundMotionSim.C

```diff
@@ -0,0 +1,152 @@
+/*************************************************/
+/*           DO NOT MODIFY THIS HEADER           */
+/*                                               */
+/*                     MASTODON                  */
+/*                                               */
+/*    (c) 2015 Battelle Energy Alliance, LLC     */
+/*            ALL RIGHTS RESERVED                */
+/*                                               */
+/*   Prepared by Battelle Energy Alliance, LLC   */
+/*     With the U. S. Department of Energy       */
+/*                                               */
+/*     See COPYRIGHT for full restrictions       */
+/*************************************************/
+
+#include "GroundMotionSim.h"
+registerMooseObject("MastodonApp", GroundMotionSim);
+
+
+InputParameters
+GroundMotionSim::validParams()
+{
+  InputParameters params = PiecewiseLinearBase::validParams();
+  params.addRequiredParam<Real>("M", "Magnitude of Event");
+  params.addRequiredParam<Real>("R", "Joyner-Boore or Epicentral Distance");
+  params.addRequiredParam<Real>("Vs30", "Shear Wave Velocity at 30m");
+  MooseEnum F("SS TF NF", "SS");
+  MooseEnum R_type("Rjb Repi", "Rjb");
+  params.addParam<MooseEnum>("F",F,"Style of Faulting");
+  params.addParam<MooseEnum>("R_type",R_type,"Indicates whether R distance is Joyner-Boore or Epicentral distance");
+  params.addParam<unsigned int>("n",1,"Number of Simulations to run for each set of inputs");
+  params.set<std::vector<Real>>("xy_data") = {};
+  params.addClassDescription(
+      "Calculates a ground acceleration history using a ground motion model written in a python script");
+  return params;
+}
+
+
+// Define arbitrarily large size of preallocated arrays for t and Acc
+#define BUFSIZE 20000
+
+GroundMotionSim::GroundMotionSim(const InputParameters & parameters) : PiecewiseLinearBase(parameters)
+{
+  const auto M = this->template getParam<Real>("M");
+  const auto R = this->template getParam<Real>("R");
+  const auto V = this->template getParam<Real>("Vs30");
+  const auto n = this->template getParam<unsigned int>("n");
+  std::string R_type = this->template getParam<MooseEnum>("R_type");
+  std::string F = this->template getParam<MooseEnum>("F");
+
+  // Convert input parameters to strings
+  std::string Mstr = std::to_string(M);
+  std::string Rstr = std::to_string(R);
+  std::string Vstr = std::to_string(V);
+  std::string nstr = std::to_string(n);
+  
+  // Concatenate command with input parameters
+  std::string command = "python $HOME/projects/mastodon/python/mastodonutils/GMSim.py GroundMotionSim ";
```

Comment:
You might want to make the python executable a parameter that defaults to `python` (or `python3` if your code requires this). Some older systems still default python to python2.